### ISSUE TITLE
Build failure: bitwarden

### ISSUE THREAD

**Issue by FreaxMATE** [NixOS/nixpkgs#264793]
### Steps To Reproduce

1. build bitwarden

### Build log

```bash
[kunruh@nixos:~]$ nix-shell -p bitwarden --show-trace 
error:
       … while evaluating the attribute 'buildInputs' of the derivation 'shell'

       at /nix/store/z707kdjl26j97153sc62szg61sz5ki34-nixos-23.05/nixos/pkgs/stdenv/generic/make-derivation.nix:303:7:

          302|     // (lib.optionalAttrs (attrs ? name || (attrs ? pname && attrs ? version)) {
          303|       name =
             |       ^
          304|         let

       … while evaluating the attribute 'installPhase' of the derivation 'bitwarden-2023.4.0'

       at /nix/store/z707kdjl26j97153sc62szg61sz5ki34-nixos-23.05/nixos/pkgs/stdenv/generic/make-derivation.nix:303:7:

          302|     // (lib.optionalAttrs (attrs ? name || (attrs ? pname && attrs ? version)) {
          303|       name =
             |       ^
          304|         let

       … while evaluating the attribute 'handled'

       at /nix/store/z707kdjl26j97153sc62szg61sz5ki34-nixos-23.05/nixos/pkgs/stdenv/generic/check-meta.nix:447:7:

          446|       # or, alternatively, just output a warning message.
          447|       handled =
             |       ^
          448|         {

       … while evaluating the attribute 'no'

       at /nix/store/z707kdjl26j97153sc62szg61sz5ki34-nixos-23.05/nixos/pkgs/stdenv/generic/check-meta.nix:449:11:

          448|         {
          449|           no = handleEvalIssue { inherit meta attrs; } { inherit (validity) reason errormsg; };
             |           ^
          450|           warn = handleEvalWarning { inherit meta attrs; } { inherit (validity) reason errormsg; };

       … while evaluating call site

       at /nix/store/z707kdjl26j97153sc62szg61sz5ki34-nixos-23.05/nixos/pkgs/stdenv/generic/check-meta.nix:449:16:

          448|         {
          449|           no = handleEvalIssue { inherit meta attrs; } { inherit (validity) reason errormsg; };
             |                ^
          450|           warn = handleEvalWarning { inherit meta attrs; } { inherit (validity) reason errormsg; };

       … while calling 'handleEvalIssue'

       at /nix/store/z707kdjl26j97153sc62szg61sz5ki34-nixos-23.05/nixos/pkgs/stdenv/generic/check-meta.nix:226:38:

          225|
          226|   handleEvalIssue = { meta, attrs }: { reason , errormsg ? "" }:
             |                                      ^
          227|     let

       error: Package ‘electron-24.8.6’ in /nix/store/z707kdjl26j97153sc62szg61sz5ki34-nixos-23.05/nixos/pkgs/development/tools/electron/binary/generic.nix:27 is marked as insecure, refusing to evaluate.


       Known issues:
        - Electron version 24.8.6 is EOL

       You can install it anyway by allowing this package, using the
       following methods:

       a) To temporarily allow all insecure packages, you can use an environment
          variable for a single invocation of the nix tools:

            $ export NIXPKGS_ALLOW_INSECURE=1

        Note: For `nix shell`, `nix build`, `nix develop` or any other Nix 2.4+
        (Flake) command, `--impure` must be passed in order to read this
        environment variable.

       b) for `nixos-rebuild` you can add ‘electron-24.8.6’ to
          `nixpkgs.config.permittedInsecurePackages` in the configuration.nix,
          like so:

            {
              nixpkgs.config.permittedInsecurePackages = [
                "electron-24.8.6"
              ];
            }

       c) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
          ‘electron-24.8.6’ to `permittedInsecurePackages` in
          ~/.config/nixpkgs/config.nix, like so:

            {
              permittedInsecurePackages = [
                "electron-24.8.6"
              ];
            }
```

### Notify maintainers

@amarshall 
@Kiwi 

### Metadata

```bash
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.57, NixOS, 23.05 (Stoat), 23.05.4335.898cb2064b6e`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.5`
 - channels(kunruh): `""`
 - channels(root): `"nixos-23.05"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

```

**Comment by FliegendeWurst** [comment]
The build failure is due to: `Electron version 24.8.6 is EOL`

The error message itself gives some guidance on how to proceed.

**Comment by FliegendeWurst** [comment]
Besides that, there is a PR to upgrade to a newer Electron version #264492

**Comment by eclairevoyant** [comment]
fix has been backported via #264472
